cachedb/redis.c

```diff
@@ -249,16 +250,37 @@ redis_lookup(struct module_env* env, struct cachedb_env* cachedb_env,
 
 static void
 redis_store(struct module_env* env, struct cachedb_env* cachedb_env,
-	char* key, uint8_t* data, size_t data_len)
+	char* key, uint8_t* data, size_t data_len, uint64_t ttl)
 {
 	redisReply* rep;
-	char cmdbuf[4+(CACHEDB_HASHSIZE/8)*2+3+1]; /* "SET " + key + " %b" */
 	int n;
+	int size;
+	bool set_ttl = !env->cfg->serve_expired || env->cfg->serve_expired_ttl > 0;
 
-	verbose(VERB_ALGO, "redis_store %s (%d bytes)", key, (int)data_len);
+        if (env->cfg->serve_expired_ttl > 0) {
+		ttl += env->cfg->serve_expired_ttl;
+	}
+
+	if (!set_ttl) {
+		size = 4+(CACHEDB_HASHSIZE/8)*2+3+1;
+	}
+	else {
+		size = 4+(CACHEDB_HASHSIZE/8)*2+3+4+sizeof(uint64_t)+1;
+	}
+
+	char cmdbuf[size]; /* "SET " + key + " %b EX " + ttl */
```

Comment:
This should move uptop with the rest of the declarations in this code block (function) to make the code more portable (pre-std99 complain about mixed declarations and code).